scripts/repoint-frontend/repoint_frontend.py

```diff
@@ -0,0 +1,166 @@
+        "    {% endif %}",
+    ]
+    return path, match_block, insertion_lines
+
+
+def get_config_alternate(path, environment):
+    match_block = [
+        "<LoadBalancer>",
+        "<Server name=\"{{ TARGET_SERVER_OVERRIDE | default('communications-manager-target') }}\"/>",
+        "</LoadBalancer>",
+        "<Path>{requestpath}</Path>",
+    ]
+    insertion_lines = [
+        "       {% if ENVIRONMENT_TYPE == 'sandbox' %}",
+        "           <LoadBalancer>",
+        "	            <Server name=\"{{ TARGET_SERVER_OVERRIDE | default('communications-manager-target') }}\"/>",
+        "           </LoadBalancer>",
+        "           <Path>{requestpath}</Path>",
+        "       {% else %}",
+        f"           <URL>https://comms-apim.{environment}.communications.national.nhs.uk</URL>",
+        "       {% endif %}",
+    ]
+    return path, match_block, insertion_lines
+
+
+def get_config_by_type(block_type, *args):
+    if block_type == "default":
+        return get_config_default(*args)
+    elif block_type == "alternate":
+        return get_config_alternate(*args)
+    else:
+        raise ValueError(f"Unknown block type: {block_type}")
+
+
+def update_file(environment, file_path, match_block, insertion_lines):
+    lines = read_file(file_path)
+    match_block = [line.strip() for line in match_block]
+
+    insertion_lines = [
+        line.replace("{environment}", environment) for line in insertion_lines
+    ]
+
+    block_found, new_lines = False, []
+    i = 0
+
+    while i < len(lines):
+        # Check if current line matches the start of match_block
+        if lines[i].strip() == match_block[0]:
+            # Check if the full match_block matches line-by-line
+            match = all(
+                lines[i + j].strip() == match_block[j] for j in range(len(match_block))
+            )
+            if match:
+                block_found = True
+                i += len(match_block)
+                new_lines.extend(
+                    f"{line}\n" for line in insertion_lines
+                )  # Insert new lines
+                continue
+        new_lines.append(lines[i])
+        i += 1
+
+    if not block_found:
+        print(f"Error: Block not found in the file at {file_path}.")
+        return
+
+    write_file(file_path, new_lines)
+    print(f"Lines successfully added after the specified block in {file_path}.")
+
+
+if __name__ == "__main__":
+    # Ensure script is run from the correct directory
+    if os.path.basename(os.getcwd()) != "communications-manager-api":
+        print("Error: repoint_frontend must be run from the project root.")
+        sys.exit(1)
+
+    parser = argparse.ArgumentParser(
+        description="Script to update proxy configuration files based on the environment argument."
+    )
+    parser.add_argument(
+        "environment", type=str, help="Environment identifier (e.g. 'de-gith1')"
+    )
+    args = parser.parse_args()
+
+    # Process `default` block type configs
```

Comment:
Could we get a little more illustrative comments? I'm finding it a little hard to parse what a "config" means in the context of `get_config_by_type` and what `match_block` and `insertion_lines` are all for. For a little stand-alone script like this that's ok, and I'm glad I'm not looking at the same thing that's implemented in bash. I just hoped maybe you could help walk the reader through the intention behind what's happening in plain english comments if it's not super clear from the code on its own?